guides/working-with-git.md

```diff
@@ -0,0 +1,31 @@
+# Working With Git - Best Practices
+
+## Commiting Code
+
+Follow 3 simple rules:
+
+1. Commit locally to the _working_ branch as often as you save.
+1. Push your _working_ branch at the end of each "development session" (before walking away from your computer for lunch).
+1. Squash _working_ branch commits when merging into a primary or secondary branch.
+
+## Branching
+
+Use [GitFlow](https://datasift.github.io/gitflow/IntroducingGitFlow.html) branching model.
```

Comment:
Just a personal preference, but [GitHub Flow](https://guides.github.com/introduction/flow/) is much easier to learn and more management for teams. It's possible that there are too many concurrent releases going on at ESDC for it to be pragmatic, but suggesting anyway. 😄

Reply:
I would actually like to recommend the GitHub Flow over GitFlow because of the simplicity to it. My concerns are, most projects have at least 2 concurrent releases, and continuous delivery isn't an option yet for 95% of our projects which GitHub Flow works best with. This also has to work for projects that will remain on our TFS infrastructure using Git.
Nevertheless, I'll leave this open for a bit to see what others think.